pkg/controllers/controller.go

```diff
@@ -49,22 +50,27 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 	}
 	// 2. Copy object for merge patch base
 	persisted := resource.DeepCopyObject()
-	// 3. Set to true to remove race condition
+	// 3. Set Defaults
+	if _, ok := resource.(apis.Defaultable); ok {
+		resource.(apis.Defaultable).SetDefaults(ctx)
+	}
+	// 4. Set to true to remove race condition
 	// TODO: remove status conditions on provisioners
-	if provisioner, ok := resource.(*v1alpha1.Provisioner); ok {
-		provisioner.StatusConditions().MarkTrue(v1alpha1.Active)
+	if _, ok := resource.(apis.ConditionManager); ok {
+		resource.(apis.ConditionManager).SetCondition(apis.Condition{
```

Comment:
I think you can do 
```
if conditionManager, ok := resource.(apis.ConditionManager); ok {
  conditionsManager.StatusConditions().MarkTrue(v1alpha1.Active)
}
```

Reply:
the knative `apis.ConditionManager` interface does not have a `StatusConditions()` method.